tests/foreman/cli/test_rhcloud_inventory.py

```diff
@@ -264,3 +266,44 @@ def test_rhcloud_external_links():
 
     :CaseLevel: System
     """
+
+
+@pytest.mark.tier3
+def test_positive_generate_all_reports_job(target_sat):
+    """Generate all reports job via foreman-rake console:
+
+    :id: a9e4bfdb-6d7c-4f8c-ae57-a81442926dd8
+
+    :customerscenario: true
+
+    :Steps:
+        1. Execute Foreman GenerateAllReportsJob via cli.
+
+    :expectedresults: Reports generation works as expected.
+
+    :BZ: 2110163
+
+    :CaseAutomation: Automated
+
+    :CaseLevel: System
+    """
+    with target_sat.session.shell() as sh:
+        sh.send('foreman-rake console')
+        time.sleep(30)  # sleep to allow time for console to open
+        sh.send('ForemanTasks.async_task(ForemanInventoryUpload::Async::GenerateAllReportsJob)')
+        time.sleep(3)  # sleep for the cmd execution
+    timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M')
+    wait_for(
+        lambda: target_sat.api.ForemanTask()
+        .search(query={'search': f'{generate_report_jobs} and started_at >= "{timestamp}"'})[0]
+        .result
+        == 'success',
+        timeout=400,
+        delay=15,
+        silent_failure=True,
+        handle_exception=True,
+    )
```

Comment:
Would suggest using `target_sat.wait_for_tasks` instead

Reply:
Just keeping it for the sake of uniformity in the test file, I can do a refactor of this for the whole file with all the tests included.